servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,573 @@
+    /**
+     * Convert variable before initialization.
+     */
+    public void pack() {
+        initParamMap = ImmutableMap.copyOf(initParamMap);
+    }
+
+    @Override
+    public Object getAttribute(String name) {
+        requireNonNull(name, "name");
+        return attributeMap.get(name);
+    }
+
+    @Override
+    public Enumeration<String> getAttributeNames() {
+        return Collections.enumeration(attributeMap.keySet());
+    }
+
+    @Override
+    public void setAttribute(String name, @Nullable Object object) {
+        requireNonNull(name);
+        if (object == null) {
+            removeAttribute(name);
+            return;
+        }
+        attributeMap.put(name,object);
+    }
+
+    @Override
+    public void removeAttribute(String name) {
+        requireNonNull(name, "name");
+        attributeMap.remove(name);
+    }
+
+    @Override
+    public String getServletContextName() {
+        return servletContextName;
+    }
+
+    @Override
+    public ServletRegistration addServlet(String servletName, String className) {
+        requireNonNull(servletName, "servletName");
+        requireNonNull(className, "className");
+        try {
+            return addServlet(servletName, (Class<HttpServlet>) Class.forName(className));
+        } catch (ClassNotFoundException e) {
+            logger.error("Add servlet failed: ", e);
+        }
+        return null;
+    }
+
+    @Override
+    public ServletRegistration addServlet(String servletName, Servlet servlet) {
+        requireNonNull(servletName, "servletName");
+        requireNonNull(servlet, "servlet");
+        checkArgument(!isInitialized(), "Can't add servlet after server is initialized");
+        if (!servletName.isEmpty() && servletName.charAt(servletName.length() - 1) == '/') {
+            servletName = servletName.substring(0, servletName.length() - 1);
+        }
+        final ServletRegistration servletRegistration;
+        servletRegistration = new ServletRegistration(servletName, servlet, this, servletUrlMapper);
+        servletRegistrationMap.put(servletName,servletRegistration);
+        servletRegistration.addMapping(servletName);
+        return servletRegistration;
+    }
+
+    @Override
+    public ServletRegistration addServlet(String servletName, Class<? extends Servlet> servletClass) {
+        requireNonNull(servletName, "servletName");
+        requireNonNull(servletClass, "servletClass");
+        try {
+            return addServlet(servletName, createServlet(servletClass));
+        } catch (ServletException e) {
+            logger.error("Add servlet failed: ", e);
+        }
+        return null;
+    }
+
+    @Override
+    public <T extends Servlet> T createServlet(Class<T> clazz) throws ServletException {
+        try {
+            requireNonNull(clazz, "clazz");
+            return clazz.newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+            logger.error("Create servlet failed: ", e);
```

Comment:
`throw new RuntimeException("Failed to create a serlvet: " + class.getSimpleName() +, e)`